src/nl/rubensten/texifyidea/intentions/LatexLabelDefiningNewCommand.kt

```diff
@@ -0,0 +1,112 @@
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import nl.rubensten.texifyidea.psi.LatexCommands
+import nl.rubensten.texifyidea.settings.LabelingCommandInformation
+import nl.rubensten.texifyidea.settings.TexifySettings
+import nl.rubensten.texifyidea.settings.labeldefiningcommands.EditLabelDefiningCommand
+import nl.rubensten.texifyidea.util.*
+
+open class LatexLabelDefiningNewCommand : TexifyIntentionBase("Add label defining command to list") {
+    private val settings = TexifySettings.getInstance()
+
+    override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?): Boolean {
+        if (editor == null || file == null || !file.isLatexFile()) {
+            return false
+        }
+
+        // check if selected part is \label
+        val element = file.findElementAt(editor.caretModel.offset) ?: return false
+        var selected = element as? LatexCommands ?: element.parentOfType(LatexCommands::class) ?: return false
+
+        // get parent element
+        val parentElement = selected.parent
+        val parent = parentElement as? LatexCommands
+                ?: parentElement.parentOfType(LatexCommands::class)
+
+        // when element is \label and parent is \newcommand check if command is already in list
+        if (selected.name == "\\label" && parent?.name == "\\newcommand") {
+            // get name of the defined command
+            val cmdName = parent.requiredParameter(0) ?: return false
+            // check if there is already an entry wih this name
+            return !settings.labelCommands.containsKey(cmdName)
+        }
+
+        // check if the parent of the current position is \newcommand, if true, set the selected command to the parent
+        // command
+        if (parent?.name == "\\newcommand") {
+            selected = parent
+        }
+
+        // when command is \newcommand, check if it contains a \label
+        if (selected.name == "\\newcommand") {
+            val children = selected.childrenOfType(LatexCommands::class)
+            if (children.none { it.name == "\\label" }) {
+                return false
+            }
+            val cmdName = selected.requiredParameter(0) ?: return false
+            return !settings.labelCommands.containsKey(cmdName)
+        }
+
+        return false
+    }
+
+    override fun invoke(project: Project, editor: Editor?, file: PsiFile?) {
+        if (editor == null || file == null || !file.isLatexFile()) {
+            return
+        }
+
+        // check if selected part is \label
+        val element = file.findElementAt(editor.caretModel.offset) ?: return
+        val selected = element as? LatexCommands ?: element.parentOfType(LatexCommands::class) ?: return
+
+        // get parent element
+        val parentElement = selected.parent
+        val parent = parentElement as? LatexCommands
+                ?: parentElement.parentOfType(LatexCommands::class)
```

Comment:
Can be a oneliner.

Reply:
fixed in commit 4fc09f4f03cfda54b13164fb3b1d6e31c694f878